gateway-service/src/test/java/org/zowe/apiml/gateway/zaas/ZaasControllerTest.java

```diff
@@ -0,0 +1,193 @@
+import org.mockito.Mock;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.zowe.apiml.gateway.security.service.schema.source.AuthSource;
+import org.zowe.apiml.gateway.security.service.schema.source.AuthSourceService;
+import org.zowe.apiml.gateway.security.service.schema.source.ParsedTokenAuthSource;
+import org.zowe.apiml.gateway.security.service.zosmf.ZosmfService;
+import org.zowe.apiml.message.core.MessageService;
+import org.zowe.apiml.message.yaml.YamlMessageService;
+import org.zowe.apiml.passticket.IRRPassTicketGenerationException;
+import org.zowe.apiml.passticket.PassTicketService;
+
+import java.util.Date;
+
+import static org.apache.http.HttpStatus.*;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+class ZaasControllerTest {
+
+    @Mock
+    private AuthSourceService authSourceService;
+
+    @Mock
+    private PassTicketService passTicketService;
+
+    @Mock
+    private ZosmfService zosmfService;
+
+    private MockMvc mockMvc;
+
+    private JSONObject ticketBody;
+
+    private AuthSource.Parsed authSource;
+
+    private static final String PASSTICKET_URL = "/gateway/zaas/ticket";
+    private static final String ZOSMF_TOKEN_URL = "/gateway/zaas/zosmf";
+
+    private static final String USER = "test_user";
+    private static final String PASSTICKET = "test_passticket";
+    private static final String APPLID = "test_applid";
+    private static final String JWT_TOKEN = "jwt_test_token";
+    private static final String LTPA_TOKEN = "ltpa_test_token";
+
+    @BeforeEach
+    void setUp() throws IRRPassTicketGenerationException, JSONException {
+        MessageService messageService = new YamlMessageService("/gateway-messages.yml");
+
+        when(passTicketService.generate(anyString(), anyString())).thenReturn(PASSTICKET);
+        ZaasController zaasController = new ZaasController(authSourceService, messageService, passTicketService, zosmfService);
+        mockMvc = MockMvcBuilders.standaloneSetup(zaasController).build();
+        ticketBody = new JSONObject()
+            .put("applicationName", APPLID);
+    }
+
+    @Nested
+    class GivenAuthenticated {
+
+        @BeforeEach
+        void setUp() {
+            authSource = new ParsedTokenAuthSource(USER, new Date(111), new Date(222), AuthSource.Origin.ZOSMF);
+        }
+
+        @Test
+        @Disabled
```

Comment:
Why new tests are disabled? whenRequestZoweToken_thenResonseOK and whenRequestingZosmfTokens_thenInternalServerError as well